config/src/test/scala/com/typesafe/config/impl/ConfigSubstitutionTest.scala

```diff
@@ -924,20 +924,24 @@ class ConfigSubstitutionTest extends TestUtils {
 
     @Test
     def substSelfReferenceIndirect() {
+        // this has two possible outcomes depending on whether
+        // we resolve and memoize a first or b first. currently
+        // java 8's hash table makes it resolve OK, but
```

Comment:
i agree it's a pathological situation that shouldn't happen often. still i doubt LinkedHashMap has so much overhead that we should avoid using it. my vote would still be for the more deterministic behavior over slight performance gain, but i'll let you make the call on it. I think that `avoidDelayedMergeObjectResolveProblem5` also depends on map ordering, btw